### DISCUSSION TITLE
Is it possible to specify nested path for discriminated unions?

### DISCUSSION THREAD

**Discussion by mn-prp** [colinhacks/zod#2082] [Q&A] [answered]
I have a schema that allows for multiple object types, all of which have a `type` property. Unfortunately, the `type` property is not a primitive, it is an object with structure `{ key: string, label: string }`, where the key is the unique id and the label is a human-readable version of it. Is there a way that I can use Zod so that I can take advantage of disriminated unions without needing to pre-process and change the structure of my data?

**Comment by JacobWeisenburger** [comment, marked as answer]
Currently, I don't think there is a way to do this. Sorry about that. Perhaps you could submit a PR to fix this.
